### PR TITLE
[WFLY-18892]:Update for jdk21

### PR DESCRIPTION
**Pull request by jimma**

### CODE REVIEW

File: jakarta-xml-binding/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Reference: https://docs.docker.com/engine/reference/builder/
 
-FROM fedora:35
+FROM fedora:39
 RUN ["dnf", "upgrade", "--refresh", "-y"]
-RUN ["dnf", "install", "-y", "bash-completion", "ant", "java-11-openjdk", "java-11-openjdk-devel", "java-11-openjdk-jmods", "java-17-openjdk", "java-17-openjdk-devel", "java-17-openjdk-jmods", "java-latest-openjdk", "java-latest-openjdk-devel", "java-latest-openjdk-jmods", "maven", "maven-openjdk11", "git", "git-extras", "wget", "unzip", "procps-ng", "file"]
```
Comment (scottmarlow) [inline jakarta-xml-binding/Dockerfile:5]: I think we should add Java 21 JDK as is done by this pull request but also keep adding Java 11/17.

File: jakarta-xml-binding/README.adoc [hunk 2]
```diff
@@ -44,13 +44,11 @@ Once the container is running you can execute the TCK with `run-tck.sh`. This wi
 While the current default is Java 11, you can easily set the version with the `javaenv` or `javahome` command. Simply
 pass the version of Java you want to use. The following are a list of the current installed versions:
 
-- 11
-- 17
-- 18
```
Comment (scottmarlow) [inline jakarta-xml-binding/README.adoc:49]: I think Java 11/17 should still be kept.

File: jakarta-xml-binding/Dockerfile (hunk 1 above)
Comment (jamezp) [reply jakarta-xml-binding/Dockerfile:5]: +1 we should keep both 11 and 17.

File: jakarta-xml-binding/README.adoc (hunk 2 above)
Comment (jamezp) [reply jakarta-xml-binding/README.adoc:49]: +1

**Review by jamezp — CHANGES_REQUESTED** [review]

File: jakarta-xml-binding/Dockerfile (hunk 1 above)
Comment (jimma) [reply jakarta-xml-binding/Dockerfile:5]: So we don't create a separate branch for running tck with java21 ?  After "-DnoSecurityManager=true" is added, it is all running without security manager.

**Review by scottmarlow — APPROVED** [review]

**Comment by scottmarlow** [comment]
Looks good to me, thanks @jimma!

**Review by jamezp — APPROVED** [review]

**Pull request merged**